lib/Sema/CSSolver.cpp

```diff
@@ -74,8 +74,7 @@ Solution ConstraintSystem::finalize() {
 
   // Update the best score we've seen so far.
   auto &ctx = getASTContext();
-  assert(ctx.TypeCheckerOpts.DisableConstraintSolverPerformanceHacks ||
-         !solverState->BestScore || CurrentScore <= *solverState->BestScore);
+  assert(!solverState->BestScore || CurrentScore <= *solverState->BestScore);
```

Comment:
@xedin i noticed during a recent rebase & build that the `ctx` var is now marked unused after this change. was the new `performanceHacksEnabled()` method intended to continue to be consulted in this assertion, or is it safe to delete the now-unused var?